### ISSUE TITLE
Additional issue with click events

### ISSUE THREAD

**Issue by acasar** [octobercms/ajax#6]
In the old framework we used to be able to do something like this (a very simplified example):

```html
<form>
    <input type="hidden" name="input" value="1">
    <a href="#" id="target" data-request="onTestHandler"></a>
</form>

<script>
    $('#target').data('request-data', 'additional:1').trigger('click');
</script>
```

However in the new framework ``trigger('click')`` does not trigger the request any more.

**Comment by daftspunk** [comment]
Hi @acasar 

This is something to be aware of when using jQuery: jQuery event triggers will not trigger vanilla JS events. More on this topic can be found here:

https://makandracards.com/makandra/71890-events-triggered-by-jquery-cannot-be-observed-by-native-event-listeners

Important excerpt:

> This is currently a WONTFIX for jQuery (see closed issues [#2476](https://github.com/jquery/jquery/issues/2476) , [#3347](https://github.com/jquery/jquery/issues/3347) ).

With rationale:

> jQuery doesn't trigger native events, only the ones that it tracks. Native events don't provide a way to determine when they're attached.

It's not a huge issue but some code will require native events to be used. Something like this:

```
$('#target').data('request-data', 'additional:1').get(0).click();
```

We can't write an adapter for this because the listener does listen for both and so this is a problem for supporting both.

**Comment by daftspunk** [comment]
BTW - This code should probably use `request()` instead of `click()` to trigger the request directly:

```
$('#target').data('request-data', 'additional:1').request();
```

An alternative solution that should work...

**Comment by acasar** [comment]
Hi @daftspunk! I tried both options and they both trigger the request correctly, however the additional data is not included in the request. To make it work I had to use the ``attr`` function instead of the ``data`` function. So the working solution is:

```js
$('#target').attr('data-request-data', 'additional:1').request();
```

I'm not quite sure why we need to modify the attribute directly in this scenario, but this is the only way I got it to work. Anyway, for me this is a perfectly acceptable solution, so if you do not plan any other changes, feel free to close the issue.
